testing/src/main/java/io/grpc/testing/TestUtils.java

```diff
@@ -76,7 +76,10 @@ public static ServerInterceptor recordRequestHeadersInterceptor(
   /**
    * Capture the request attributes. Useful for testing ServerCalls.
    * {@link ServerCall#getAttributes()}
+   *
+   * @deprecated Not for public use
```

Comment:
are you intending to remove this?

Reply:
Yes, can not think of a usecase of it for testing from user's perspective.